sqswatcher/plugins/sge.py

```diff
@@ -118,6 +154,12 @@ def removeHost(hostname,cluster_user):
     command = ("/opt/sge/bin/lx-amd64/qconf -dattr hostgroup hostlist %s @allhosts" % hostname)
     __runSgeCommand(command)
 
+    # Reschedule any jobs that are on the host
```

Comment:
You probably want to move this before line 145. The qmod manpage says `qmod -rj` requires admin-host privileges. I'm trying to find the relevant portion of the gridengine source code to verify that this is truly the case.

Reply:
The code I've been running in production for the last couple months actually had the entire "Removing host as administrative host" stanza removed, not just relocated into removeHost() as you had done.

I don't trust moving the rescheduling before the all.q and allhosts changes, though, since that could allow jobs to sneak on to the node. So if you think the admin host removal still needs to happen I'd recommend moving that piece to right after the job rescheduling.

Aside from that, though, this `qmod -rj` is running on the master node as root, so I'm not convinced it matters whether the hostname being removed is in the admin list or not. In fact, at the time this runs for many of my test cases, the node in question has already stopped running.